datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/BatchAddTagsResolver.java

```diff
@@ -52,6 +64,59 @@ public CompletableFuture<Boolean> get(DataFetchingEnvironment environment) throw
     });
   }
 
+  /**
+   * When adding a tag to a schema field in the UI, there's a chance the parent entity has siblings.
+   * If the given urn doesn't have a schema or doesn't have the given column, we should try to add the
+   * tag to one of its siblings. If that fails, keep trying all siblings until one passes or all fail.
+   * Then we throw if none succeed.
```

Comment:
If someone adds... lets say 2 tags, will this work?

If not, should it?

![Screen Shot 2023-02-01 at 3 17 50 PM](https://user-images.githubusercontent.com/17549204/216189251-c971dac2-a9eb-49e9-bc78-93a13a1bea97.png)

Reply:
yup it'll work just fine! same thing with multiple glossary terms.

The reason I do the check to ensure there's only 1 target urn is for the case where people are bulk adding tags/terms to entities. If we want to check for siblings with multiple entities this becomes a good amount more complicated as the logic for everything is very separated where it would be tough to check for siblings for each one.

also in the UI it's not possible to truly bulk add with schema fields, so I think it's a pretty safe assumption. Things might fail if someone tries to bulk add to multiple schema fields at once through our API but I think that's alright.

So that helper method should actually be renamed from `handleAddSingleTagToSchemaField` -> `handleAddTagsToSingleSchemaField`

Reply:
makes sense!